arrows/vxl/color_commonality_filter.cxx

```diff
@@ -0,0 +1,561 @@
+                     d->settings.grid_resolution_height,
+                     "Divide the height of the image into x regions, if enabled." );
+  config->set_value( "grid_resolution_width",
+                     d->settings.grid_resolution_width,
+                     "Divide the width of the image into x regions, if enabled." );
+
+  return config;
+}
+
+// ----------------------------------------------------------------------------
+void
+color_commonality_filter
+::set_configuration( vital::config_block_sptr in_config )
+{
+  // Start with our generated vital::config_block to ensure that assumed values
+  // are present. An alternative would be to check for key presence before
+  // performing a get_value() call.
+  vital::config_block_sptr config = this->get_configuration();
+  config->merge_config( in_config );
+
+  // Settings for ccf
+  d->color_resolution_per_chan =
+    config->get_value< unsigned >( "color_resolution_per_channel",
+                                   d->color_resolution_per_chan );
+  d->intensity_resolution =
+    config->get_value< unsigned >( "intensity_resolution",
+                                   d->intensity_resolution );
+  d->settings.output_scale_factor =
+    config->get_value< unsigned >( "output_scale",
+                                   d->settings.output_scale_factor );
+  d->settings.grid_image =
+    config->get_value< bool >( "grid_image",
+                               d->settings.grid_image );
+  d->settings.grid_resolution_width =
+    config->get_value< unsigned >( "grid_resolution_width",
+                                   d->settings.grid_resolution_width );
+  d->settings.grid_resolution_height =
+    config->get_value< unsigned >( "grid_resolution_height",
+                                   d->settings.grid_resolution_height );
+
+  d->color_resolution = d->color_resolution_per_chan *
+                        d->color_resolution_per_chan *
+                        d->color_resolution_per_chan;
+
+  d->color_histogram.resize( d->color_resolution, 0 );
+  d->intensity_histogram.resize( d->intensity_resolution, 0 );
+}
+
+// ----------------------------------------------------------------------------
+bool
+color_commonality_filter
+::check_configuration( vital::config_block_sptr config ) const
+{
+  if( !is_power_of_two( d->color_resolution_per_chan ) ||
+      !is_power_of_two( d->intensity_resolution ) )
+  {
+    LOG_ERROR( logger(), "Specified resolutions must be a power of 2" );
+  }
+  return true;
+}
+
+// ----------------------------------------------------------------------------
+kwiver::vital::image_container_sptr
+color_commonality_filter
+::filter( kwiver::vital::image_container_sptr image_data )
+{
+  // Perform Basic Validation
+  if( !image_data )
+  {
+    return kwiver::vital::image_container_sptr();
+  }
+
+  if( image_data->depth() != 1 && image_data->depth() != 3 )
+  {
+    LOG_ERROR( logger(), "Invalid number of input planes!" );
```

Comment:
Would "unsupported" be better here? (Instead of "invalid".)

Same for the later exception.